mitodl/mitxonline
# 842: sync-coursrun-upgrade-deadline-with-edx

#### Pre-Flight checklist

#### What are the relevant tickets?
https://github.com/mitodl/mitxonline/issues/842

#### What's this PR do?
Adds additional steps to the existing `sync_courseruns_data` celery task in order to update CourseRun's `upgrade_deadline` within `mitxonline` with information from edX's Course Mode API endpoint.

#### How should this be manually tested?
The way I tested this locally was:
You must have `devstack` running locally in order to complete the test below.

1. After checking out the branch associated with this PR, continue to the next steps.
2. Checkout the branch from `edx-api-client` here: https://github.com/mitodl/edx-api-client/pull/87
3. Copy the `edx-api` directory from the project in step 1 and paste it in the root directory or your `mitxonline` project.
4. Remove `edx-api-client==1.1.0` and `edx-api-client==1.1.0` from `requirements.in` and `requirements.txt` in the root of your `mitxonline` project.
5. Perform a `docker-compose build` from the root of your `mitxonline` project.
6. Inside your `mitxonline` project, open the `main/settings.py` and modify the `sync-courseruns-data` `minute`, `hour`, and `day_of_week` to all equal `*`.  This will cause the celery task to execute every minute.
7. Ensure that you have the `OPENEDX_SERVICE_WORKER_API_TOKEN` configured in `mitxonline` and also defined matching a record in `devstack`
8. Perform `docker-compose up` from the root of your `mitxonline` project.
9. Verify that `mitxonline` and specifically the `celery` service start up without any issues.
10. Ensure that you have a configured Course Run in `mitxonline` which matches a Course in `devstack` but does not define an `upgrade_deadline` value.
11. Navigate to http://localhost:18000/admin/course_modes/coursemode/
12. Create a Course Mode record that defines an upgrade deadline and corresponds with the Course Run you created in step 10.
13. Watch the celery logs for `Updated upgrade deadline for course run <COURSE RUN ID HERE>`.
14. Within the `mitxonline` Django Admin, verify that the Course Run record from step 10 now has a value for `upgrade_deadline` that matches the value defined in step 12.
